satpy/writers/geotiff.py

```diff
@@ -34,7 +34,13 @@ class GeoTIFFWriter(ImageWriter):
 
         >>> scn.save_datasets(writer='geotiff')
 
-    Un-enhanced float geotiff with NaN for fill values:
+    By default the writer will use the :class:`~satpy.writers.Enhancer` class to
+    linear stretch the data (see :doc:`../enhancements`).
+    To get Un-enhanced images ``enhance=False`` can be specified which will
+    write a geotiff with the data type of the dataset. The fill value defaults
+    to the the datasets ``"fill_value"`` attribute if not ``None`` and no value is
```

Comment:
`_FillValue` not `fill_value`, it's a CF convention thing.

Reply:
True for CF conventions its `_FillValue` but as far as I can see, at least for the geotiff writer, neither the `_FillValue` nor `fill_value` attribute is used. But I think that could easily be implemented.

Reply:
See https://github.com/pytroll/trollimage/blob/master/trollimage/xrimage.py#L970-L995

Reply:
Ok I see. I didn't think of looking in xrimage since for floats the NaN gets set in the geotiff writer if `fill_value` is not passed and I didn't find any reference to `_FillValue` around there. Just one question why isn't `_FillValue` checked for float data? Wouldn't it be more consistent to default to NaN for float data only if `_FillValue` is None or non existent and no `fill_value` parameter is passed similar to the integer data case?

Reply:
I suppose it could be but in most cases a float having the `_FillValue` attribute is usually a mistake in the reader code where it wasn't removed. The `_FillValue` usually corresponds to the in-file integer fill value and not the in-memory unscaled version of the data. I guess I wouldn't be against it, but I also don't think it would be a very common case. It is also very likely that any xarray operations will put NaNs into the float data for invalid values.

Reply:
Ok yes from a "complete" Satpy workflow point of view starting with a reader this makes totally sense since `_FillValue` should indeed be removed. I guess I thought it would in general be easier to understand for the user how fill values area handled if the order of what is checked is similar between data types. But I agree this is probably not a very common case (especially for possible workflows I have in mind) and in any case would also just shift the "problem" from the user supplying ``fill_value`` parameter to setting the `_FillValue` attribute and I guess in any case, as most of the time, the user should know what he's doing anyway. I updated the documentation and hope it is clearer now. Sorry for all the back and forth about this rather "unimportant" detail.